qiskit/visualization/latex.py

```diff
@@ -345,690 +317,319 @@ def _get_beamer_page(self):
-        return qindex
+                    if op.condition:
+                        self._add_condition(op, wire_list, column)
 
-    def _ffs(self, mask):
-        """Find index of first set bit.
+                    if len(wire_list) == 1:
+                        self._latex[wire_list[0]][column] = "\\gate{%s}" % gate_text
 
-        Args:
-            mask (int): integer to search
-        Returns:
-            int: index of the first set bit.
-        """
-        origin = (mask & (-mask)).bit_length()
-        return origin - 1
+                    elif len(wire_list) == 2:
+                        if op.name == "swap":
+                            self._build_swap(wire_list, column)
 
+                        elif op.name == 'rzz' or base_name in ['u1', 'p', 'rzz']:
+                            symm_name = 'rzz' if op.name == 'rzz' else base_name
+                            num_cols_used = self._build_symmetric_gate(op, symm_name, gate_text,
+                                                                       wire_list, column,
+                                                                       num_cols_used)
+                        elif isinstance(op.op, ControlledGate):
+                            self._build_single_ctrl_gate(op, gate_text, wire_list, column)
 
-def _get_register_specs(bits):
-    """Get the number and size of unique registers from bits list.
+                        else:
+                            self._build_multi_gate(gate_text, wire_list, column)
 
-    Args:
-        bits (list[Bit]): this list is of the form::
-            [Qubit(v0, 0), Qubit(v0, 1), Qubit(v0, 2), Qubit(v0, 3), Qubit(v1, 0)]
-            which indicates a size-4 register and a size-1 register
+                    elif isinstance(op.op, ControlledGate):
+                        num_cols_used = self._build_multi_ctrl_gate(op, gate_text,
+                                                                    wire_list, column,
+                                                                    num_cols_used)
+                    else:
+                        self._build_multi_gate(gate_text, wire_list, column)
 
-    Returns:
-        OrderedDict: ordered map of Registers to their sizes
-    """
-    regs = collections.OrderedDict([(bit.register, bit.register.size) for bit in bits])
-    return regs
+            column += num_cols_used
 
+    def _build_multi_gate(self, gate_text, wire_list, col):
+        """Add a multiple wire gate to the _latex list"""
+        wire1 = min(wire_list)
+        wire2 = max(wire_list)
+        self._latex[wire1][col] = "\\multigate{%s}{%s}" % \
+            (wire2 - wire1, gate_text)
+        for wire in range(wire1 + 1, wire2 + 1):
+            self._latex[wire][col] = "\\ghost{%s}" % gate_text
+
+    def _build_single_ctrl_gate(self, op, gate_text, wire_list, col):
+        """Add a gate with a single control to the _latex list"""
+        wire1 = min(wire_list)
+        wire2 = max(wire_list)
```

Comment:
Same as in https://github.com/Qiskit/qiskit-terra/pull/5855/files#r589716077

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/5855/commits/9fbb01f272d9b61d5dd96c47c449f4a67272f484.